### DISCUSSION TITLE
netcdf/pnetcdf not installing netcdf_par.h

### DISCUSSION THREAD

**Discussion by richgilham** [Homebrew/discussions#5445] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.6
ORIGIN: https://github.com/Homebrew/brew
HEAD: e8430b25a1d1321f32e5093d62b57b5cb7cfb3c3
Last commit: 2 days ago
Core tap JSON: 18 Jun 10:40 UTC
Core cask tap JSON: 18 Jun 10:40 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 11
Homebrew Ruby: 3.3.3 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.3/bin/ruby
CPU: 11-core 64-bit arm_lobos
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.6.0 => /usr/bin/curl
macOS: 14.5-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

New brew user...

I've used brew to get the dependencies to build XIOS. However, the build falls over having not found netcdf_par.h. I have both netcdf and pnetcdf formulae installed and can find netcdf.h in a sensible location but not the _par variation.

Me problem or brew problem?

Many thanks,

Rich

**Comment by oliviermarti** [comment]
I'm having the same issue. Any hint ?

**Comment by carlocab** [comment]
Using `netcdf_par.h` requires building `netcdf` with MPI support[^1], which is currently not provided by the `netcdf` package.

[^1]: For future references, this also requires passing `-DNETCDF_ENABLE_PARALLEL4=ON` to `cmake`. See https://github.com/Unidata/netcdf-c/blob/49ca8441c34bbd86ba5286c1efb38968c541db08/include/CMakeLists.txt#L38.

**Reply by jhaiduce** [reply]
> Using `netcdf_par.h` requires building `netcdf` with MPI support, which is currently not provided by the `netcdf` package.

The question was about the `pnetcdf` package, which _does_ provide MPI support according to its [description](https://formulae.brew.sh/formula/pnetcdf):

> Parallel netCDF library for scientific data using the OpenMPI library

**Reply by gromgit** [reply]
> The question was about the `pnetcdf` package, which _does_ provide MPI support

But not `netcdf_par.h`:-
```bash
% tar tvf pnetcdf-1.14.1.tar.gz | grep netcdf_par.h
%
```
I don't use either package, but I suspect that PnetCDF is _not_ a drop-in replacement for NetCDF-with-MPI.
